test/test_torch.py

```diff
@@ -3729,6 +3729,16 @@ def test_cat(self):
 
         self.assertRaises(RuntimeError, lambda: torch.cat([]))
 
+    def test_cat_padding(self):
+        x = torch.rand(13, 1)
+        y = torch.rand(17, 2)
+        z = torch.rand(19, 3)
+
+        res1 = torch.cat((x, y, z), 0, 0)
+        self.assertEqual(res1.narrow(0, 0, 13).narrow(1, 0, 1), x, 0)
+        self.assertEqual(res1.narrow(0, 13, 17).narrow(1, 0, 2), y, 0)
+        self.assertEqual(res1.narrow(0, 30, 19).narrow(1, 0, 3), z, 0)
```

Comment:
I'd also like to see a test of different dimensionality, if that's supported.

Reply:
As I mentioned before, whatever pad_value is, all inputs should have the same dim().